app-tests/run.sh

```diff
@@ -0,0 +1,150 @@
+#!/bin/bash
+set -e
+
+export OPAL_AUTH_PUBLIC_KEY
+export OPAL_AUTH_PRIVATE_KEY
+export OPAL_AUTH_MASTER_TOKEN
+export OPAL_CLIENT_TOKEN
+export OPAL_DATA_SOURCE_TOKEN
+
+function generate_opal_keys {
+  echo "- Generating OPAL keys"
+
+  ssh-keygen -q -t rsa -b 4096 -m pem -f opal_crypto_key -N ""
+  OPAL_AUTH_PUBLIC_KEY="$(cat opal_crypto_key.pub)"
+  OPAL_AUTH_PRIVATE_KEY="$(tr '\n' '_' < opal_crypto_key)"
+  rm opal_crypto_key.pub opal_crypto_key
+
+  OPAL_AUTH_MASTER_TOKEN="$(openssl rand -hex 16)"
+  OPAL_AUTH_JWT_AUDIENCE=https://api.opal.ac/v1/ OPAL_AUTH_JWT_ISSUER=https://opal.ac/ OPAL_REPO_WATCHER_ENABLED=0 opal-server run &
+  sleep 2;
+  OPAL_CLIENT_TOKEN="$(opal-client obtain-token "$OPAL_AUTH_MASTER_TOKEN" --type client)"
+  OPAL_DATA_SOURCE_TOKEN="$(opal-client obtain-token "$OPAL_AUTH_MASTER_TOKEN" --type datasource)"
+  # shellcheck disable=SC2009
+  ps -ef | grep opal-server | grep -v grep | awk '{print $2}' | xargs kill
+  sleep 5;
+
+  echo "- Create .env file"
+  rm -f .env
+  (
+    echo "OPAL_AUTH_PUBLIC_KEY=\"$OPAL_AUTH_PUBLIC_KEY\"";
+    echo "OPAL_AUTH_PRIVATE_KEY=\"$OPAL_AUTH_PRIVATE_KEY\"";
+    echo "OPAL_AUTH_MASTER_TOKEN=\"$OPAL_AUTH_MASTER_TOKEN\"";
+    echo "OPAL_CLIENT_TOKEN=\"$OPAL_CLIENT_TOKEN\"";
+    echo "OPAL_AUTH_PRIVATE_KEY_PASSPHRASE=\"$OPAL_AUTH_PRIVATE_KEY_PASSPHRASE\""
+  ) > .env
+}
+
+function prepare_policy_repo {
+  echo "- Clone tests policy repo to create test's branch"
+  export POLICY_REPO_BRANCH
+  POLICY_REPO_BRANCH=test-$RANDOM$RANDOM
+  rm -rf ./opal-tests-policy-repo
+  git clone git@github.com:permitio/opal-tests-policy-repo.git
+  cd opal-tests-policy-repo
+  git checkout -b $POLICY_REPO_BRANCH
+  git push --set-upstream origin $POLICY_REPO_BRANCH
+  cd -
+
+  # That's for the docker-compose to use, set ssh key from "~/.ssh/id_rsa", unless another path/key data was configured
+  export OPAL_POLICY_REPO_SSH_KEY
+  OPAL_POLICY_REPO_SSH_KEY_PATH=${OPAL_POLICY_REPO_SSH_KEY_PATH:-~/.ssh/id_rsa}
+  OPAL_POLICY_REPO_SSH_KEY=${OPAL_POLICY_REPO_SSH_KEY:-$(cat "$OPAL_POLICY_REPO_SSH_KEY_PATH")}
```

Comment:
don't you need to remove new lines or something like that ?

Reply:
Apparently not. I think the double quotes take care of that